tools/test/examples/README.md

```diff
@@ -0,0 +1,81 @@
+# Examples testing script
+
+examples.py is script which makes easier to test examples.
+```
+usage: examples.py [-h] [-c CONFIG] [-e EXAMPLE] [-p TEST_PHASE]
+                   {import,clone,deploy,tag,compile,export} ...
+
+positional arguments:
+  {import,clone,deploy,tag,compile,export}
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -c CONFIG             Configuration file in json. examples.json as default
+  -e EXAMPLE, --example EXAMPLE
+                        filter the examples that are used in the script
+  -p TEST_PHASE, --test-phase TEST_PHASE
+                        Name of test phase which configuration is used.
+                        nightly_test is default
+```
+
+ Configuration example:
+ ```
+ {
+  "examples": [
+    {
+      "name": "mbed-os-example-blinky",
+      "github": "https://github.com/ARMmbed/mbed-os-example-blinky",
+      "mbed": ["https://os.mbed.com/teams/mbed-os-examples/code/mbed-os-example-blinky"],
+      "test-repo-source": "github",
+      "features": [],
+      "baud_rate": 9600,
+      "compare_log": ["mbed-os-example-blinky/tests/blinky.log"],
+      "auto-update": true,
+      "pull_request_test": {
+        "targets": ["K64F"],
+        "toolchains": ["GCC_ARM"],
+        "exporters": [],
+        "compile": true,
+        "export": true,
+        "test": false
+      },
+      "nightly_test": {
+        "targets": ["K64F", "K66F"],
+        "toolchains": ["GCC_ARM", "ARM"],
+        "exporters": [],
+        "compile": true,
+        "export": true,
+        "test": true
+      },
+      "release_test": {
+        "targets": ["K64F", "K66F", "NUCLEO_F429ZI"],
+        "toolchains": [],
+        "exporters": ["IAR"],
+        "compile": true,
+        "export": true,
+        "test": true
+      }
+    }
+  ]
+}
+ ```
+ Fields documentation:
+ * **name** is example name used mostly with prints
+ * **github** is github repository address
+ * **mbed** Mercurial repositories in os.mbed.com
+ * **test-repo-source** specify which repo source to use for testing
+ * **features***
+ * **baud_rate** used baud rate
+ * **compare_log** Log which is compared to command line output when testing
+ * Testing phases:
+   * **pull_request_test** This configuration is used for every pull request testing
+   * **nightly_test** This is used in nightly test
```

Comment:
I guess we can't use inheritance - all tehse are separately defined so will maintain set of 3 (PR nightly and release)

Reply:
We had long discussion about this with @adbridge and decided not to use inheritance because it will make it more complex and harder to understand.